keras_cv/layers/regularization/dropblock_2d.py

```diff
@@ -0,0 +1,206 @@
+
+    Usage:
+    DropBlock2D can be used inside a `tf.keras.Model`:
+    ```python
+    # (...)
+    x = Conv2D(32, (1, 1))(x)
+    x = BatchNormalization()(x)
+    x = ReLU()(x)
+    x = DropBlock2D(0.1, dropblock_size=7)(x)
+    # (...)
+    ```
+    When used directly, the layer will zero-out some inputs in a contiguous region and
+    normalize the remaining values.
+
+    ```python
+    # Small feature map shape for demonstration purposes:
+    features = tf.random.stateless_uniform((1, 4, 4, 1), seed=[0, 1])
+
+    # Preview the feature map
+    print(features[..., 0])
+    # tf.Tensor(
+    # [[[0.08216608 0.40928006 0.39318466 0.3162533 ]
+    #   [0.34717774 0.73199546 0.56369007 0.9769211 ]
+    #   [0.55243933 0.13101244 0.2941643  0.5130266 ]
+    #   [0.38977218 0.80855536 0.6040567  0.10502195]]], shape=(1, 4, 4),
+    # dtype=float32)
+
+    layer = DropBlock2D(0.1, dropblock_size=2, seed=1234) # Small size for demonstration
+    output = layer(features, training=True)
+
+    # Preview the feature map after dropblock:
+    print(output[..., 0])
+    # tf.Tensor(
+    # [[[0.10955477 0.54570675 0.5242462  0.42167106]
+    #   [0.46290365 0.97599393 0.         0.        ]
+    #   [0.7365858  0.17468326 0.         0.        ]
+    #   [0.51969624 1.0780739  0.80540895 0.14002927]]], shape=(1, 4, 4),
+    # dtype=float32)
+
+    ```
+    We can observe two things:
+    1. A 2x2 block has been set to zero.
+    2. The inputs have been normalized.
+
+    One must remember, that DropBlock operation is random, so bigger or smaller
+    patches can be dropped.
+    """
+
+    def __init__(
+        self,
+        dropout_rate,
+        dropblock_size,
+        data_format=None,
+        seed=None,
+        name=None,
+    ):
+        super().__init__(seed=seed, name=name, force_generator=True)
+        if not 0.0 <= dropout_rate <= 1.0:
+            raise ValueError(
+                f"dropout_rate must be a number between 0 and 1. "
+                f"Received: {dropout_rate}"
+            )
+
+        self._dropout_rate = dropout_rate
+        self._dropblock_height, self._dropblock_width = conv_utils.normalize_tuple(
+            value=dropblock_size, n=2, name="dropblock_size", allow_zero=False
+        )
+        self._data_format = conv_utils.normalize_data_format(data_format)
+
+    def call(self, x, training=None):
+        if not training or self._dropout_rate == 0.0:
+            return x
+
+        if self._data_format == "channels_last":
+            _, height, width, _ = tf.split(tf.shape(x), 4)
+        else:
+            _, _, height, width = tf.split(tf.shape(x), 4)
+
+        # Unnest scalar values
+        height = tf.squeeze(height)
+        width = tf.squeeze(width)
+
+        dropblock_height = tf.math.minimum(self._dropblock_height, height)
+        dropblock_width = tf.math.minimum(self._dropblock_width, width)
+
+        # Seed_drop_rate is the gamma parameter of DropBlock.
+        seed_drop_rate = (
```

Comment:
I am little bit confused about this value. Is it the scale up factor for the output value? eg should it be 
"size_input / (size_ input - size_dropout)"? (this value will be * to all the undropped pixel in the output)

Reply:
This is the `gamma` parameter from the paper:
![gamma](https://user-images.githubusercontent.com/58402418/158750467-d0d0de6b-e3ad-4e69-9ca9-54f3ebd3b38c.png)

This is taken directly from [original TF implementation](https://github.com/tensorflow/tpu/blob/b24729de804fdb751b06467d3dce0637fa652060/models/official/detection/modeling/architecture/nn_ops.py#L234), but modified so that it works with non-square inputs.

Reply:
@sebastian-sz wouldn't be best to just call this `gamma` instead of `seed_drop_rate`?